### PR TITLE
Improve Facebook workflow

### PR DESCRIPTION
**Pull request by holiber**
- allow streaming to groups
- allow streaming to the user's timeline
- allow selecting a scheduled event for streaming
- the vue-multiselect library has been updated (I needed to update this library because the new version allows better customization)
- game selectors for twitch and facebook are different inputs now
- the twitch game selector shows a game image
- CustomResolution input has improved UX now
- streaming tests have been sliced by several files
- added a `privacy` parameter for FB streams

fbdestination
re-connectfb

### CODE REVIEW

File: app/components/windows/ShareStream.tsx [hunk 1]
```diff
@@ -127,7 +127,8 @@ export default class ShareStream extends TsxComponent<{ sharePageUrl: string }>
   }
 
   get sharePageUrl() {
-    return `facebook.com/${this.facebookService.state.activePage.name}-${this.facebookService.state.activePage.id}`;
+    return '';
+    // return `facebook.com/${this.facebookService.state.activePage.name}-${this.facebookService.state.activePage.id}`;
```
Comment (gettinToasty) [inline app/components/windows/ShareStream.tsx:131]: i thought that we got rid of this page entirely?

File: app/components/windows/go-live/PlatformSettings.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ export default class PlatformSettings extends TsxComponent<Props> {
     return (
       <ValidatedForm class="flex" ref="settingsForm">
         <div style={{ width: '100%' }}>
-          {!hasPlatforms && $t('Enable at least one destination to start streaming')}
+          {!hasPlatforms && $t('Enable at least one destinationType to start streaming')}
```
Comment (gettinToasty) [inline app/components/windows/go-live/PlatformSettings.tsx:61]: i feel like we should not use camelCase in any user-facing copy

File: app/components/windows/go-live/ScheduleStreamWindow.tsx [hunk 3]
```diff
@@ -190,7 +190,7 @@ export default class ScheduleStreamWindow extends TsxComponent<{}> {
                 <HFormGroup metadata={this.formMetadata.time} vModel={this.startTimeModel.time} />
               </div>
             )}
-            {shouldShowWarn && <div>{$t('Select at least one destination')}</div>}
+            {shouldShowWarn && <div>{$t('Select at least one destinationType')}</div>}
```
Comment (gettinToasty) [inline app/components/windows/go-live/ScheduleStreamWindow.tsx:193]: same as above comment

File: app/services/platforms/mixer.ts [hunk 4]
```diff
@@ -186,7 +186,7 @@ export class MixerService extends BasePlatformService<IMixerServiceState>
     ).then(json => json.viewersCurrent);
   }
 
-  async putChannelInfo({ title, game }: IMixerStartStreamOptions): Promise<boolean> {
+  async putChannelInfo({ title, game }: IMixerStartStreamOptions): Promise<void> {
```
Comment (gettinToasty) [inline app/services/platforms/mixer.ts:189]: I think technically speaking we could just get rid of this whole service right?

File: app/services/platforms/twitch.ts [hunk 5]
```diff
@@ -279,14 +281,28 @@ export class TwitchService extends BasePlatformService<ITwitchServiceState>
...
+      data: { id: string; name: string; box_art_url: string }[];
+    }>('twitch', `https://api.twitch.tv/helix/games?name=${name}`);
+    return gamesResponse.data.map(g => {
+      const imageTemplate = g.box_art_url;
+      const imageSize = this.gameImageSize;
+      const image = imageTemplate
+        .replace('{width}', imageSize.width.toString())
+        .replace('{height}', imageSize.height.toString());
+      return { id: g.id, name: g.name, image };
+    })[0];
```
Comment (gettinToasty) [inline app/services/platforms/twitch.ts:305]: love to see us switching over to the new twitch api whenever possible

**Review by gettinToasty — APPROVED** [review]
overall seems to mostly make sense, had a few non-blocking questions

File: app/components/LiveDock.vue.ts [hunk 6]
```diff
@@ -65,7 +66,7 @@ export default class LiveDock extends Vue {
     this.underlyingSelectedChat = val;
   }
 
-  viewStreamTooltip = $t('Go to YouTube to view your live stream');
+  viewStreamTooltip = $t('Open browser to view your live stream');
```
Comment (andycreeth) [inline app/components/LiveDock.vue.ts:69]: Let's change this to "View your live stream in a web browser"

File: app/components/windows/ShareStream.tsx [hunk 7]
```diff
@@ -75,7 +75,7 @@ export default class ShareStream extends TsxComponent<{ sharePageUrl: string }>
   };
 
   mounted() {
-    this.facebookService.sendPushNotif();
+    // this.facebookService.sendPushNotif();
```
Comment (andycreeth) [inline app/components/windows/ShareStream.tsx:78]: Delete?

File: app/components/windows/ShareStream.tsx (hunk 1 above)
Comment (andycreeth) [reply app/components/windows/ShareStream.tsx:131]: Yeah I'm also curious where this component is being used.  Can we delete?

File: app/components/windows/go-live/DestinationSwitchers.tsx [hunk 8]
```diff
@@ -88,7 +88,7 @@ export class DestinationSwitchers extends TsxComponent<Props> {
             {isPrimary ? (
               <span
                 vTooltip={$t(
-                  'You cannot disable the platform you used to sign in to Streamlabs OBS. Please sign in with a different platform to disable streaming to this destination.',
+                  'You cannot disable the platform you used to sign in to Streamlabs OBS. Please sign in with a different platform to disable streaming to this destinationType.',
```
Comment (andycreeth) [inline app/components/windows/go-live/DestinationSwitchers.tsx:91]: Lol - this looks like a find-replace error.  I assume you didn't mean to change this string.

File: app/components/windows/go-live/DestinationSwitchers.tsx [hunk 9]
```diff
@@ -114,7 +114,7 @@ export class DestinationSwitchers extends TsxComponent<Props> {
   }
 
   /**
-   * Renders a switcher for a custom destination
+   * Renders a switcher for a custom destinationType
```
Comment (andycreeth) [inline app/components/windows/go-live/DestinationSwitchers.tsx:117]: Also looks like a find-replace error

File: app/components/windows/go-live/PlatformSettings.tsx (hunk 2 above)
Comment (andycreeth) [reply app/components/windows/go-live/PlatformSettings.tsx:61]: Looks like another find-replace error

File: app/components/windows/go-live/ScheduleStreamWindow.tsx (hunk 3 above)
Comment (andycreeth) [reply app/components/windows/go-live/ScheduleStreamWindow.tsx:193]: Another one :)

File: app/components/windows/go-live/platforms/FacebookEditStreamInfo.tsx [hunk 10]
```diff
@@ -1,59 +1,417 @@
...
+          <HFormGroup title={this.formMetadata.group.title}>
+            <ListInput
+              vModel={fbSettings.groupId}
+              metadata={this.formMetadata.group}
+              handleOpen={() => this.loadPictures('group')}
+              showImagePlaceholder={true}
+              imageSize={{ width: 44, height: 44 }}
+            />
+            <p>
+              {$t('Make sure Streamlabs app is added to your Group.')}
```
Comment (andycreeth) [inline app/components/windows/go-live/platforms/FacebookEditStreamInfo.tsx:326]: "Make sure the Streamlabs app is added to your Group."

**Review by andycreeth — CHANGES_REQUESTED** [review]

File: app/components/shared/inputs/BaseFormGroup.ts [hunk 11]
```diff
@@ -31,11 +31,13 @@ export default abstract class BaseFormGroup extends BaseInput<any, IInputMetadat
...
     const inputMetadata = options;
 
-    // FormGroup handle the render of the FormInput title
-    // so remove the title from FormInput metadata
-    delete inputMetadata.title;
-    delete inputMetadata.tooltip;
-    delete inputMetadata.description;
+    if (options.type === 'bool') {
+      // FormGroup handle the render of the FormInput title
+      // so remove the title from FormInput metadata
```
[the remark this answers is not in the document]
Comment (holiber) [reply app/components/shared/inputs/BaseFormGroup.ts:36]: Only bool inputs show their title and tooltip independently from FormGroup

File: app/components/windows/ShareStream.tsx (hunk 1 above)
Comment (holiber) [reply app/components/windows/ShareStream.tsx:131]: We can get rid of this component

File: app/components/windows/go-live/PlatformSettings.tsx (hunk 2 above)
Comment (holiber) [reply app/components/windows/go-live/PlatformSettings.tsx:61]: My IDE did a bad job with replacing =(

File: app/services/platforms/mixer.ts (hunk 4 above)
Comment (holiber) [reply app/services/platforms/mixer.ts:189]: Yes, we can. I prefer to not do it as a part of this PR

**Pull request merged**